hawkular-bus-common/src/main/java/org/hawkular/bus/common/consumer/RPCBasicMessageListener.java

```diff
@@ -117,8 +116,12 @@ public void onMessage(Message message) {
                 if (session == null) {
                     msglog.errorNoSessionInListener();
                 } else {
-                    producerContext.setMessageProducer(session.createProducer(replyTo));
+                    MessageProducer sessionProducer = session.createProducer(replyTo);
+
+                    producerContext.setMessageProducer(sessionProducer);
                     sender.send(producerContext, responseBasicMessage);
+
+                    sessionProducer.close();
```

Comment:
shouldn't this be done in the finally block rather than here? If send() throws an exception, you'll never close this.